node/node.go

```diff
@@ -1619,62 +1664,183 @@ func (app *App) stopServices(ctx context.Context) {
-	filename := filepath.Join(app.Config.SMESHING.Opts.DataDir, edKeyFileName)
-	app.log.Info("Looking for identity file at `%v`", filename)
+// TestIdentity loads a pre-configured identity for testing purposes.
+func (app *App) TestIdentity() ([]*signing.EdSigner, error) {
+	if len(app.Config.TestConfig.SmesherKey) == 0 {
+		return nil, nil
+	}
 
-	var data []byte
-	if len(app.Config.TestConfig.SmesherKey) > 0 {
-		app.log.With().Error("!!!TESTING!!! using pre-configured smesher key")
-		data = []byte(app.Config.TestConfig.SmesherKey)
-	} else {
-		var err error
-		data, err = os.ReadFile(filename)
+	app.log.With().Error("!!!TESTING!!! using pre-configured smesher key")
+	dst, err := hex.DecodeString(app.Config.TestConfig.SmesherKey)
+	if err != nil {
+		return nil, fmt.Errorf("decoding private key: %w", err)
+	}
+	if len(dst) != signing.PrivateKeySize {
+		return nil, fmt.Errorf("invalid key size %d/%d", dst, signing.PrivateKeySize)
+	}
+	signer, err := signing.NewEdSigner(
+		signing.WithPrivateKey(dst),
+		signing.WithPrefix(app.Config.Genesis.GenesisID().Bytes()),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to construct identity from data file: %w", err)
+	}
+
+	app.log.With().Info("Loaded testing identity", signer.PublicKey())
+	return []*signing.EdSigner{signer}, nil
+}
+
+// MigrateExistingIdentity migrates the legacy identity file to the new location.
+//
+// The legacy identity file is expected to be located at `app.Config.SMESHING.Opts.DataDir/key.bin`.
+//
+// TODO(mafa): this can be removed in a future version when the legacy identity file is no longer expected to exist.
+func (app *App) MigrateExistingIdentity() error {
+	oldKey := filepath.Join(app.Config.SMESHING.Opts.DataDir, legacyKeyFileName)
+	app.log.Info("Looking for legacy identity file at `%v`", oldKey)
+
+	src, err := os.Open(oldKey)
+	if err != nil {
+		if os.IsNotExist(err) {
+			app.log.Info("Legacy identity file not found.")
+			return nil
+		}
+		return fmt.Errorf("failed to read legacy identity file: %w", err)
+	}
+	defer src.Close()
+
+	newKey := filepath.Join(app.Config.DataDir(), keyDir, defaultKeyFileName)
+	if err := os.MkdirAll(filepath.Dir(newKey), 0o700); err != nil {
+		return fmt.Errorf("failed to create directory for identity file: %w", err)
+	}
+
+	dst, err := os.Create(newKey)
+	if err != nil {
+		return fmt.Errorf("failed to create new identity file: %w", err)
+	}
+	defer dst.Close()
+
+	if _, err := io.Copy(dst, src); err != nil {
+		return fmt.Errorf("failed to copy identity file: %w", err)
+	}
+
+	if err := src.Close(); err != nil {
+		return fmt.Errorf("failed to close legacy identity file: %w", err)
+	}
+
+	if err := os.Rename(oldKey, oldKey+".bak"); err != nil {
+		return fmt.Errorf("failed to rename legacy identity file: %w", err)
+	}
+
+	app.log.Info("Migrated legacy identity file to `%v`", newKey)
+	return nil
+}
+
+// LoadIdentities loads all existing identities from the config directory.
+func (app *App) LoadIdentities() error {
```

Comment:
Would it make sense to move all identities-related code to `node/identities.go`. This file is already huge and difficult to follow.